l1-contracts/contracts/bridge/ntv/L2NativeTokenVault.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {BeaconProxy} from "@openzeppelin/contracts-v4/proxy/beacon/BeaconProxy.sol";
+import {IBeacon} from "@openzeppelin/contracts-v4/proxy/beacon/IBeacon.sol";
+import {UpgradeableBeacon} from "@openzeppelin/contracts-v4/proxy/beacon/UpgradeableBeacon.sol";
+
+import {IERC20} from "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+
+import {INativeTokenVault} from "./INativeTokenVault.sol";
+import {IL2NativeTokenVault} from "./IL2NativeTokenVault.sol";
+import {NativeTokenVault} from "./NativeTokenVault.sol";
+
+import {IL2SharedBridgeLegacy} from "../interfaces/IL2SharedBridgeLegacy.sol";
+import {BridgedStandardERC20} from "../BridgedStandardERC20.sol";
+
+import {DEPLOYER_SYSTEM_CONTRACT, L2_ASSET_ROUTER_ADDR} from "../../common/L2ContractAddresses.sol";
+import {L2ContractHelper, IContractDeployer} from "../../common/libraries/L2ContractHelper.sol";
+
+import {SystemContractsCaller} from "../../common/libraries/SystemContractsCaller.sol";
+import {DataEncoding} from "../../common/libraries/DataEncoding.sol";
+
+import {EmptyAddress, EmptyBytes32, AddressMismatch, DeployFailed, AssetIdNotSupported} from "../../common/L1ContractErrors.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @notice The "default" bridge implementation for the ERC20 tokens. Note, that it does not
+/// support any custom token logic, i.e. rebase tokens' functionality is not supported.
+contract L2NativeTokenVault is IL2NativeTokenVault, NativeTokenVault {
+    using SafeERC20 for IERC20;
+
+    /// @dev Chain ID of L1 for bridging reasons.
+    uint256 public immutable L1_CHAIN_ID;
+
+    IL2SharedBridgeLegacy public immutable L2_LEGACY_SHARED_BRIDGE;
+
+    /// @dev Bytecode hash of the proxy for tokens deployed by the bridge.
+    bytes32 internal l2TokenProxyBytecodeHash;
+
+    /// @notice Initializes the bridge contract for later use.
+    /// @param _l1ChainId The L1 chain id differs between mainnet and testnets.
+    /// @param _l2TokenProxyBytecodeHash The bytecode hash of the proxy for tokens deployed by the bridge.
+    /// @param _aliasedOwner The address of the governor contract.
+    /// @param _legacySharedBridge The address of the L2 legacy shared bridge.
+    /// @param _bridgedTokenBeacon The address of the L2 token beacon for legacy chains.
+    /// @param _contractsDeployedAlready Ensures beacon proxy for standard ERC20 has not been deployed.
+    /// @param _wethToken Address of WETH on deployed chain
+    constructor(
+        uint256 _l1ChainId,
+        address _aliasedOwner,
+        bytes32 _l2TokenProxyBytecodeHash,
+        address _legacySharedBridge,
+        address _bridgedTokenBeacon,
+        bool _contractsDeployedAlready,
+        address _wethToken,
+        bytes32 _baseTokenAssetId
```

Comment:
this will require amendmends to CTM as well as changes to the `GatewayUpgrade`

Reply:
Why?

Reply:
base token asset id is unique for each chain, i.e. the part where we currently only fetch the shared legacy bridge has to be extended to also grab the token asset id